copy/copy.go

```diff
@@ -668,10 +663,15 @@ func (ic *imageCopier) copyLayer(ctx context.Context, srcInfo types.BlobInfo, ba
 		}
 		if reused {
 			logrus.Debugf("Skipping blob %s (already present):", srcInfo.Digest)
+			bar := ic.c.createProgressBar(pool, srcInfo, "blob", "completed: already exists")
+			bar.SetTotal(0, true)
 			return blobInfo, cachedDiffID, nil
 		}
 	}
 
+	bar := ic.c.createProgressBar(pool, srcInfo, "blob", "completed")
+	defer bar.SetTotal(srcInfo.Size, true)
```

Comment:
Can we just do this on the success path? Marking the progress as “everything succeeded” on the error paths seems conceptually incorrect.

Reply:
Makes perfect sense, good catch!